composer/datasets/cifar_hparams.py

```diff
@@ -166,3 +168,25 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataLoaderHpara
                                                     batch_size=batch_size,
                                                     sampler=sampler,
                                                     drop_last=self.drop_last)
+
+
+@dataclass
+class StreamingCIFAR10Hparams(DatasetHparams):
+    """Streaming CIFAR10 hyperparameters.
+
+    Args:
+        remote (str): Remote directory (S3 or local filesystem) where dataset is stored.
+        local (str): Local filesystem directory where dataset is cached during operation.
+    """
+
+    remote: str = hp.optional('Remote directory (S3 or local filesystem) where dataset is stored',
+                              default='s3://mosaicml-internal-dataset-cifar10/mds/1/')
+    local: str = hp.optional('Local filesystem directory where dataset is cached during operation',
+                             default='/tmp/mds-cache/mds-cifar10/')
+
+    def initialize_object(self, batch_size: int, dataloader_hparams: DataLoaderHparams) -> DataLoader:
+        dataset = StreamingCIFAR10(self.remote, self.local, self.is_train, self.shuffle, batch_size)
```

Comment:
Could we use named args like `StreamingCIFAR10(remote=self.remote, local=self.local, ...)`? I get nervous because sometimes we add new things like retries, maybe a new transform arg, etc.

Reply:
Done